examples/LiftSim_baseline/A2C/actor.py

```diff
@@ -0,0 +1,137 @@
+from copy import deepcopy
+from lift_model import LiftModel
+from lift_agent import LiftAgent
+
+
+@parl.remote_class
+class Actor(object):
+    def __init__(self, config):
+        self.config = config
+        self.env_num = config['env_num']
+
+        self.envs = []
+        for _ in range(self.env_num):
+            env = LiftSim()
+            env = RewardWrapper(env)
+            env = ActionProcessWrapper(env)
+            env = ObsProcessWrapper(env)
+            env = MetricsWrapper(env)
+            self.envs.append(env)
+        self.vector_env = VectorEnv(self.envs)
+
+        # number of elevators
+        self.ele_num = self.envs[0].mansion_attr.ElevatorNumber
+
+        act_dim = self.envs[0].act_dim
+        self.obs_dim = self.envs[0].obs_dim
+        self.config['obs_dim'] = self.obs_dim
+
+        # nested list of shape (env_num, ele_num, obs_dim)
+        self.obs_batch = self.vector_env.reset()
+        # (env_num * ele_num, obs_dim)
+        self.obs_batch = np.array(self.obs_batch).reshape(
+            [self.env_num * self.ele_num, self.obs_dim])
+
+        model = LiftModel(act_dim)
+        algorithm = parl.algorithms.A3C(
+            model, vf_loss_coeff=config['vf_loss_coeff'])
+        self.agent = LiftAgent(algorithm, config)
+
+    def sample(self):
+        sample_data = defaultdict(list)
+
+        env_sample_data = {}
+        # treat each elevator in Liftsim as an independent env
+        for env_id in range(self.env_num * self.ele_num):
+            env_sample_data[env_id] = defaultdict(list)
+
+        for i in range(self.config['sample_batch_steps']):
+            actions_batch, values_batch = self.agent.sample(self.obs_batch)
+
+            vector_actions = np.array_split(actions_batch, self.env_num)
+            assert len(vector_actions[-1]) == self.ele_num
+            next_obs_batch, reward_batch, done_batch, info_batch = \
+                    self.vector_env.step(vector_actions)
+
+            # (env_num, ele_num, obs_dim) -> (env_num * ele_num, obs_dim)
+            next_obs_batch = np.array(next_obs_batch).reshape(
+                [self.env_num * self.ele_num, self.obs_dim])
+            # repeat reward and done to ele_num times
+            # (env_num) -> (env_num, ele_num) -> (env_num * ele_num)
+            reward_batch = np.repeat(reward_batch, self.ele_num)
+            done_batch = np.repeat(done_batch, self.ele_num)
+
+            for env_id in range(self.env_num * self.ele_num):
+                env_sample_data[env_id]['obs'].append(self.obs_batch[env_id])
+                env_sample_data[env_id]['actions'].append(
+                    actions_batch[env_id])
+                env_sample_data[env_id]['rewards'].append(reward_batch[env_id])
+                env_sample_data[env_id]['dones'].append(done_batch[env_id])
+                env_sample_data[env_id]['values'].append(values_batch[env_id])
+
+                # Calculate advantages when the episode is done or reach max sample steps.
```

Comment:
Calculate advantages when the episode is done or `reaches` max sample steps.

Reply:
done